system/ee/ExpressionEngine/Service/File/Upload.php

```diff
@@ -468,7 +468,12 @@ public function validateFile(FileModel $file)
         $action = ($file->isNew()) ? 'upload_filedata' : 'edit_file_metadata';
 
         $file->set($_POST);
-        $file->title = (ee()->input->post('title')) ?: $file->file_name;
+
+        if (ee()->config->item('use_orig_name_as_fallback_title') === 'y') {
+            $file->title = (ee()->input->post('title')) ?: $_FILES['file']['name'];
```

Comment:
This code is assuming the file has been uploaded as field with name `file` - which is not always true. I think it needs to be more abstract to work in all cases.

Reply:
@intoeetive , I'm having a hard time to find my way around the code. Maybe the [`finishUpload()`](https://github.com/ExpressionEngine/ExpressionEngine/blob/96307e89a6702aecd70f88e6a258c7d5d7e78423/system/ee/ExpressionEngine/Controller/Files/Files.php#L240) or [`upload()`](https://github.com/ExpressionEngine/ExpressionEngine/blob/96307e89a6702aecd70f88e6a258c7d5d7e78423/system/ee/ExpressionEngine/Controller/Files/Files.php#L147) functions of the [**Files** controller](https://github.com/ExpressionEngine/ExpressionEngine/blob/6.dev/system/ee/ExpressionEngine/Controller/Files/Files.php)... I've no idea.

Reply:
I had a look at the code and it seems this function is only being called when uploading the file via Files controller. Which does not seem right, but it is what it is. And it's already using some hard-coded POST variable names.

So it's probably fine to use `$_FILES['file']` here - just make sure it's defined before using